vm/runtime/conversions.go

```diff
@@ -0,0 +1,272 @@
+	}
+	bInt, ok := tryToInt(b)
+	if !ok {
+		return 0, 0, false
+	}
+	return aInt, bInt, true
+}
+
+// tryToFloat tries converting numeric types to float64.
+func tryToFloat(v interface{}) (float64, bool) {
+	switch v := v.(type) {
+	case uint:
+		return float64(v), true
+	case uint8:
+		return float64(v), true
+	case uint16:
+		return float64(v), true
+	case uint32:
+		return float64(v), true
+	case uint64:
+		return float64(v), true
+	case int:
+		return float64(v), true
+	case int8:
+		return float64(v), true
+	case int16:
+		return float64(v), true
+	case int32:
+		return float64(v), true
+	case int64:
+		return float64(v), true
+	case time.Duration:
+		return float64(v), true
+	case float32:
+		return float64(v), true
+	case float64:
+		return v, true
+	default:
+		return 0, false
+	}
+}
+
+// tryBothToFloat tries converting both values to float64, using tryToFloat.
+func tryBothToFloat(a, b interface{}) (float64, float64, bool) {
+	aFloat, ok := tryToFloat(a)
+	if !ok {
+		return 0, 0, false
+	}
+	bFloat, ok := tryToFloat(b)
+	if !ok {
+		return 0, 0, false
+	}
+	return aFloat, bFloat, true
+}
+
+// tryToString converts the given value to string if being of string type.
+func tryToString(v interface{}) (string, bool) {
+	switch v := v.(type) {
+	case string:
+		return v, true
+	default:
+		return "", false
+	}
+}
+
+// tryBothToString tries converting both values to string, using tryToString.
+func tryBothToString(a, b interface{}) (string, string, bool) {
+	aStr, ok := tryToString(a)
+	if !ok {
+		return "", "", false
+	}
+	bStr, ok := tryToString(b)
+	if !ok {
+		return "", "", false
+	}
+	return aStr, bStr, true
+}
+
+// tryToTime converts the given value to time.Time if it is of time.Time type.
+func tryToTime(v interface{}) (time.Time, bool) {
+	switch v := v.(type) {
+	case time.Time:
+		return v, true
+	default:
+		return time.Time{}, false
+	}
+}
+
+// tryBothToTime tries converting both values to time.Time, using tryToTime.
+func tryBothToTime(a, b interface{}) (time.Time, time.Time, bool) {
+	aTime, ok := tryToTime(a)
+	if !ok {
+		return time.Time{}, time.Time{}, false
+	}
+	bTime, ok := tryToTime(b)
+	if !ok {
+		return time.Time{}, time.Time{}, false
+	}
+	return aTime, bTime, true
+}
+
+// tryToDuration converts the given value to time.Duration if being of
+// time.Duration type.
+func tryToDuration(v interface{}) (time.Duration, bool) {
+	switch v := v.(type) {
+	case time.Duration:
+		return v, true
+	default:
+		return 0, false
+	}
+}
+
+// tryBothToDuration tries converting both values to time.Duration, using
+// tryToDuration.
+func tryBothToDuration(a, b interface{}) (time.Duration, time.Duration, bool) {
+	aDur, ok := tryToDuration(a)
+	if !ok {
+		return 0, 0, false
+	}
+	bDur, ok := tryToDuration(b)
+	if !ok {
+		return 0, 0, false
+	}
+	return aDur, bDur, true
+}
+
+// typeName is a type representation, avoiding plain reflect-types.
+type typeName int
+
+// Supported type names.
+const (
+	typeUInt = iota
```

Comment:
`typeUInt typeName = iota`
is a bit nicer as they then are typed

Reply:
Absolutely yes! I forgot setting the type. Thank you.